src/main/scala/com/fulcrumgenomics/umi/CodecConsensusCaller.scala

```diff
@@ -0,0 +1,196 @@
+                           minDuplexLength: Int = 1
+                          ) extends DuplexConsensusCaller(
+  readNamePrefix      = readNamePrefix,
+  readGroupId         = readGroupId,
+  minInputBaseQuality = minInputBaseQuality,
+  qualityTrim         = false,
+  errorRatePreUmi     = errorRatePreUmi,
+  errorRatePostUmi    = errorRatePostUmi,
+  minReads            = Seq(1, 1, 1),
+  maxReadsPerStrand   = maxReadsPerStrand
+) {
+  require(this.minReadsPerStrand >= 1, "minReadsPerStrand must be at least 1.")
+
+  private val FilterDuplexTooShort = s"Top/Bottom Strand Overlap < ${minDuplexLength} bases."
+  private val FilterIndelError = "Indel error between strands in duplex overlap."
+  private val clipper = new SamRecordClipper(ClippingMode.Hard, autoClipAttributes = false)
+
+  /** Returns the MI tag as is because in CODEC there is no /A or /B unlike classic duplex-seq */
+  override protected[umi] def sourceMoleculeId(rec: SamRecord): String = rec[String]("MI")
+
+  /**
+    * Takes in all the reads for a source molecule and, if possible, generates one or more
+    * output consensus reads as SAM records.
+    *
+    * @param recs the full set of source SamRecords for a source molecule
+    * @return a seq of consensus SAM records, may be empty
+    */
+  override protected def consensusSamRecordsFromSamRecords(recs: Seq[SamRecord]): Seq[SamRecord] = {
+    val (pairs, frags) = recs.partition(_.paired)
+    rejectRecords(frags, FilterFragments)
+
+    if (pairs.isEmpty) {
+      Nil
+    }
+    else {
+      // Extract just the primary alignments and then clip where they extend past the ends of their mates
+      // TODO: Remove the isFrPair here and handle chimeric reads or reads with one unmapped etc.
+      val primaries = pairs.filterNot(r => r.secondary || r.supplementary).filter(_.isFrPair)
+      primaries.groupBy(_.name).foreach { case (_, Seq(rec, mate)) => clipper.clipExtendingPastMateEnds(rec, mate) }
+
+      val r1s = filterToMostCommonAlignment(primaries.filter(_.firstOfPair).map(toSourceReadForCodec))
+      val r2s = filterToMostCommonAlignment(primaries.filter(_.secondOfPair).map(toSourceReadForCodec))
+      if (r1s.length < this.minReadsPerStrand || r2s.length < this.minReadsPerStrand) {
+        rejectRecords((r1s.view ++ r2s.view).flatMap(_.sam), FilterMinReads)
+        Nil
+      }
+      else {
+        // Calculate the max overlap between R1 and R2, and if it's too short then filter those reads
+        val longestR1Alignment = r1s.flatMap(_.sam).maxBy(_.cigar.lengthOnTarget)
+        val longestR2Alignment = r2s.flatMap(_.sam).maxBy(_.cigar.lengthOnTarget)
+
+        val (longestPosAln, longestNegAln) =
+          if (longestR1Alignment.positiveStrand) (longestR1Alignment, longestR2Alignment)
+          else (longestR2Alignment, longestR1Alignment)
+
+        val (overlapStart, overlapEnd) = (longestNegAln.start, longestPosAln.end)
```

Comment:
question: the reads sequence towards each other still (innies), is that why we use negative for start, positive for end?  Perhaps a comment or two here?

Reply:
Yeah - I'll add a comment, but everything here is predicated on the reads being an FR pair that overlaps.